docs/core/docker/docker-basics-dotnet-core.md

```diff
@@ -0,0 +1,287 @@
+
+# copy csproj and restore as distinct layers
+COPY *.csproj ./
+RUN dotnet restore
+
+# copy and build everything else
+COPY . ./
+RUN dotnet publish -c Release -o out
+ENTRYPOINT ["dotnet", "out/Hello.dll"]
+```
+
+The Dockerfile contains Docker build instructions that run sequentially.
+
+The first instruction must be [**FROM**](https://docs.docker.com/engine/reference/builder/#from). This instruction initializes a new build stage and sets the Base Image for the remaining instructions. The multi-arch tags pull either Windows or Linux containers depending on the Docker for Windows [container mode](https://docs.docker.com/docker-for-windows/#switch-between-windows-and-linux-containers). The Base Image for our sample is the 2.0-sdk image from the microsoft/dotnet repository,
+
+```Dockerfile
+FROM microsoft/dotnet:2.0-sdk
+```
+
+The [**WORKDIR**](https://docs.docker.com/engine/reference/builder/#workdir) instruction sets the working directory for any remaining RUN, CMD, ENTRYPOINT, COPY, and ADD Dockerfile instructions. If the directory doesn't exist, it's created. In this case, WORKDIR is set to the app directory.
+
+```Dockerfile
+WORKDIR /app
+```
+
+The [**COPY**](https://docs.docker.com/engine/reference/builder/#copy) instruction copies new files or directories from the source path and adds them to the destination container filesystem. With this instruction, we are copying the C# project file to the container.
+
+```Dockerfile
+COPY *.csproj ./
+```
+
+The [**RUN**](https://docs.docker.com/engine/reference/builder/#run) instruction executes any commands in a new layer on top of the current image and commit the results. The resulting committed image is used for the next step in the Dockerfile. We are running **dotnet restore** to get the needed dependencies of the C# project file. 
+
+```Dockerfile
+RUN dotnet restore
+```
+
+This **COPY** instruction copies the rest of the files into our container into new [layers](https://docs.docker.com/engine/userguide/storagedriver/imagesandcontainers/#images-and-layers).
+
+```Dockerfile
+COPY . ./
+```
+
+We are publishing the app with this **RUN** instruction. The [**dotnet publish**](../tools/dotnet-publish.md) command compiles the application, reads through its dependencies specified in the project file, and publishes the resulting set of files to a directory. Our app is published with a **Release** configuration and output to the default directory.
+
+```Dockerfile
+RUN dotnet publish -c Release -o out
+```
+
+The [**ENTRYPOINT**](https://docs.docker.com/engine/reference/builder/#entrypoint) instruction allows the container to run as an executable.
+
+```Dockerfile
+ENTRYPOINT ["dotnet", "out/Hello.dll"]
+```
+
+Now you have a Dockerfile that:
+
+* copies your app to the container
+* your app's dependencies to the container
+* builds the app to run as an executable
+
+### Build and run the Hello .NET Core 2.0 app
+
+#### Essential Docker commands
+
+These Docker commands are essential:
```

Comment:
How was this list determined?  I am surprised a command that lists the images on disk is not included.  If the reader is a noob - learning the newish image/container/volume/etc api may be easier to pick up.  For example you can list images and containers via `docker image ls`/`docker container ls` and delete them via `docker image rm`/`docker container rm`

Reply:
This list was initially determined and approved by several different folks during the planning of these articles. I can definitely add docker image to the list. Good catch!